src/utils/word-api.ts

```diff
@@ -17,42 +30,31 @@ class WordApi {
     });
   }
 
-  static async getJabRefCitations(): Promise<Array<Word.ContentControl> | void> {
-    return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
-      await context.sync();
-      return contentControls.items.filter(
-        (citation) =>
-          citation.tag.includes("JABREF-CITATION") &&
-          !citation.tag.includes("NEW")
-      );
-    }).catch((error) => {
-      console.log(`Error: ${JSON.stringify(error)}`);
-      if (error instanceof OfficeExtension.Error) {
-        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
-      }
-    });
+  static async getJabRefCitations(
+    context: Word.RequestContext
+  ): Promise<Array<Word.ContentControl>> {
+    const { contentControls } = context.document;
+    context.load(contentControls, "tag, length");
+    await context.sync();
+    return contentControls.items.filter((citation) =>
+      citation.tag.includes("JABREF-CITATION")
+    );
   }
 
   static async getPositionOfNewCitation(): Promise<number | void> {
     return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
+      const currentPosition = context.document.getSelection();
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      const locationArray = jabRefCitations.map((citation) => {
+        const citationToCompareWith = citation.getRange("Start");
+        const currentSelectionRange = currentPosition.getRange("Whole");
+        return citationToCompareWith.compareLocationWith(currentSelectionRange);
+      });
       await context.sync();
-      let length = 0;
-      let pos = 0;
-      for (let i = 0, ilen = contentControls.items.length; i < ilen; i += 1) {
-        const { tag } = contentControls.items[i];
-        if (tag.includes("JABREF-CITATION")) {
-          if (tag.substring(16) === "NEW") {
-            pos = length;
-            break;
-          }
-          length += 1;
-        }
-      }
-      return pos;
+      const position = locationArray.findIndex(
+        (location) => location.value === "After"
+      );
+      return position !== undefined ? position : jabRefCitations.length;
```

Comment:
`position ?? jabrefCitations.length`?

Reply:
If any citation is not after the current selection, then the current selection is the last citation.

Reply:
Yes, I understand. The above was a suggestion for an equivalent code which reads a bit easier.